qiskit/primitives/base/base_estimator.py

```diff
@@ -109,11 +110,7 @@ class BaseEstimator(BasePrimitive, Generic[T]):
 
     __hash__ = None
 
-    def __init__(
-        self,
-        *,
-        options: dict | None = None,
-    ):
+    def __init__(self, *, options: dict | None = None, apply_layout: bool = True):
```

Comment:
I think this is a great feature to have, but I thought that the purpose of having an `options`dictionary as an input parameter was to allow flexibility to include new options. In this case, `apply_layout` is defined as a separate input parameter, instead of as an `option`. Is there any particular reason why? I see that later on this change forces the reference `Estimator` to expose `apply_layout`, even though it will not be used (it doesn't transpile), and this can be pretty misleading for users.

Reply:
Was any thought given to say making the operator->circuit qubit mapping more directly exposed. In the past the qubits were mapped (basis changes gates and measurements added) from operator to circuit 1:1 so circuit and operator needed to be same size. I have come across users wanting to have a larger circuit with a smaller operator, which also comes about often when transpiling along with re-ordering. Would it be more flexible to have some mapping passed through here, which could include (information from) the transpiler layout or the mapping could be more user defined if they were doing something more custom. Internally whether the operator is padded/reordered, or left alone and the mapping dealt with when adding basis gates and measurements so the measurements end up correct for the operator, can come down to the implementation. Clearly this aspect can still be done if the support is done just to take the layout from the circuit or not based on the flag - wondering whether it makes sense for the extra flexibility here. I guess by default the above is simplest in that I assume whether transpiled or not given the default value of that flag things are always going to work. Passing a mapping would be more explicit.

Reply:
@ElePT `option` is used in run, but this `option` is not an option that should be used in run.

@woodsp-ibm If users are explicitly passing a layout or mapping, users can convert it explicitly on the user's side. In this view, the implicit conversion that this PR aims for is unnecessary. However, it should be noted that there are too many errors caused by usage like the example in this PR. If the number of qubits is changing, users will notice the error, but if the number of qubits is not changing, users will not notice the error and will calculate a wrong expectation value. I am struggling with whether that is really the case.
I am very torn on whether to withdraw this PR.